README.md

```diff
@@ -1,7 +1,190 @@
+  The second formula reads "for all xs, there is a y, for all zs". Any relation coming afterwards will relate all the element x of the domain to at least an element y in the domain, which in turn is related to all elements z in the domain. 
+  (c) ∀x∃y∀z∃w and ∃x∀y∃z∀w
+  The first formula reads "for all xs, there is a y, for all zs, there is a w."
+  Any relation coming afterwards will relate all elements x in the domain with at least an element y, which in turn relates to all the elements z in the domain, which in turn are each related to at least one element w.
+``The second formula reads "there is an x, for all ys, there is a z, such that all ws".
+  Any relation coming afterwards will relate one element x in the domain with all elements y, which are each related to an element y, which in turn relate to all the elements w in the domain.
+
+7. Show that the following sentences are not equivalent by exhibiting a graph that models one but not both of these sentences:
+```
+∀x∃y∀z(R(x,y) ∧ R(x,z) ∧ R(y,z))
+∃x∀y∃z(R(x,y) ∧ R(x,z) ∧ R(y,z))
+```
+ {(a,b)(b,a)(a,a)} 
+ The second statement is true in this model. There is an x that relates every y (a, related to a and b) There is an x that relates to a z (where a is x and b is z), all ys (a and b) are related to some z (a is related to a and b, b is related to a)
+ The first statement is not true. It is not true that all xs (a and b) are related to all zs (b is not related to a)
+A graph is provided in the attached file.
+
+8. Using an online tableau proof generator - such as the one found here `https://www.umsu.de/trees/` - provide tree proofs of the following entailments, which are known as the De Morgan's laws:
+  ```
+See attached images.
+
+  (a) ∀x∀y(¬(Px ∧ Qx) → (¬Px ∨ ¬Qx))
+  (b) ∀x∀y(¬(Px ∨ Qx) → (¬Px ∧ ¬Qx))
+  (c) ∀x∀y((¬Px ∨ ¬Qx) → ¬(Px ∧ Qx))
+  (d) ∀x∀y((¬Px ∧ ¬Qx) → ¬(Px ∨ Qx))
+```
+	
+9. Using a natural deduction proof generator - such as the one found here `https://proofs.openlogicproject.org/` - provide natural deduction proofs for each of De Morgan's laws. 
+
+Using http://teachinglogic.liglab.fr/DN/index.php, since I couldn't make the other website work.
+
+a) 
+assume -(Px & Qx).
+  assume -(-Px + -Qx).
+    assume -Px.
+      -Px + -Qx.
+      F.
+    therefore --Px.
+    assume Qx.
+      Px.
+      assume -(-Px + -Qx).
+        Px & Qx.
+        F.
+      therefore --(-Px + -Qx).
+      F.
+    therefore -Qx.
+    -Px + -Qx.
+    F.
+  therefore --(-Px + -Qx).
+  -Px + -Qx.
+therefore -(Px & Qx) => -Px + -Qx.	
+
+b)
+assume -(Px + Qx).
+  assume Px.
+    Px + Qx.
+    F.
+  therefore -Px.
+  assume Qx.
+    Px + Qx.
+    F.
+  therefore -Qx.
+  -Px & -Qx.
+therefore -(Px + Qx) => -Px & -Qx.	    
+
+c)
+assume -Px + -Qx.
+  assume Px & Qx.
+    Px.
+    Qx.
+    assume -Px.
+      F.
+    therefore -Px => F.
+    assume -Qx.
+      F.
+    therefore -Qx => F.
+    F.
+  therefore -(Px & Qx).
+therefore -Px + -Qx => -(Px & Qx).

```

Comment:
This proof is very strange. We cannot get F from  -Px => F and -Qx => F. I think a correct proof is:

assume -Px + -Qx.
  (1) assume -Px
    assume Px & Qx.
      Px.
      F. (since we have -Px and Px)
    therefore -(Px & Qx)
 (2) assume -Qx
    assume Px & Qx.
      Qx.
      F. (since we have -Qx and Qx)
    therefore -(Px & Qx)  
(3) Therefore   -(Px & Qx)  [disjunctive syllogism]
 
(4) therefore -Px + -Qx => -(Px & Qx).

Reply:
@giacomodecolle Sorry but I find your proof seems to work, too! 👍